### PR TITLE
--disable-ssl-check option

### PR DESCRIPTION
**Pull request by ivanprado**
Some users are experiencing problems with the certificates on Windows. While I think the problem is that Windows certs are not up to date, anyway this is causing troubles to the users. As a scape hatch, this PR introduces the option `--disable-ssl-check`` to skip the problem by disabling SSL verification.

An alternative would have been to install our own certificates using `certifi`. But `aiohttp` guys decided [not to do so](https://github.com/aio-libs/aiohttp/issues/341) and neither should we: we should rely on OS certs instead of trying to pack the certificates ourselves.

### CODE REVIEW

**Comment by lopuhin** [comment]
Thanks @ivanprado , is the typing issue caused by changes in this PR, or happens on master as well?

**Review by lopuhin — COMMENTED** [review]
Thanks for the PR @ivanprado , c3cf87cf054ef47503ccd4906047d764c54aa8f5 looks good to me, I didn't check the other commit yet - asked about it above.

To me the main question is if we should allow it or not (as disabling certificate checks could be dangerous), but looking around I see that almost all tools do allow it as it's often not in the power of the user to fix certificate errors, so +1 to allow it for us.

File: CHANGES.rst [hunk 1]
```diff
@@ -1,6 +1,11 @@
 Changes
 =======
 
+TBR:
+------------------
+
+* --disable-ssl-check option to disable ssl verification.
```
Comment (kmike) [inline CHANGES.rst:7]: The feature looks good to me @ivanprado; the only thing is naming, as usual :) 

I know the parameter is named ssl in aiohttp, but SSL is deprecated a long time ago, and not used much by websites now (<4%); it's TLS all the way. Would it be possible to have another name for this option - tls, or something about certificates, or just something about being insecure?

**Comment by ivanprado** [comment]
> is the typing issue caused by changes in this PR, or happens on master as well?

@lopuhin It will likely happen in master as well. Tenacity introduced new type annotations (https://github.com/jd/tenacity/pull/304/files) and now `mypy` is able to check if our code matches. 

> as disabling certificate checks could be dangerous

I've been thinking a bit about it. The major issue I can see is somebody trying to do a man in the middle attack. But even with the checks on, this attack is still possible. You just need to create your own certificates (using letsencrypt for example). They will be valid. 

Note that here we are only disabling certificates validation. The communication will still be secure over TLS in any case.

File: CHANGES.rst (hunk 1 above)
Comment (ivanprado) [reply CHANGES.rst:7]: TIL TSL vs SSL. What do you think about https://github.com/zytedata/zyte-autoextract/pull/37/commits/015e5b1b57d330ac816ae0ab6ce21f484fd193b5?

**Comment by kmike** [comment]
Looks good, thanks @ivanprado!

**Pull request merged**